api-select.go

```diff
@@ -88,45 +88,140 @@ type ParquetInputOptions struct{}
 
 // CSVInputOptions csv input specific options
 type CSVInputOptions struct {
-	FileHeaderInfo       CSVFileHeaderInfo
-	RecordDelimiter      string
-	FieldDelimiter       string `xml:",omitempty"`
-	QuoteCharacter       string `xml:",omitempty"`
-	QuoteEscapeCharacter string `xml:",omitempty"`
-	Comments             string `xml:",omitempty"`
+	FileHeaderInfo       *CSVFileHeaderInfo `xml:"FileHeaderInfo"`
+	RecordDelimiter      *string            `xml:"RecordDelimiter"`
+	FieldDelimiter       *string            `xml:"FieldDelimiter"`
+	QuoteCharacter       *string            `xml:"QuoteCharacter"`
+	QuoteEscapeCharacter *string            `xml:"QuoteEscapeCharacter"`
+	Comments             *string            `xml:"Comments"`
+}
+
+// SetFileHeaderInfo sets file header info in CSV input options
+func (c *CSVInputOptions) SetFileHeaderInfo(v CSVFileHeaderInfo) {
+	c.FileHeaderInfo = &v
+}
+
+// SetRecordDelimiter sets record delimiter character in CSV input options
+func (c *CSVInputOptions) SetRecordDelimiter(v string) {
+	c.RecordDelimiter = &v
+}
+
+// SetFieldDelimiter sets field delimiter character in CSV input options
+func (c *CSVInputOptions) SetFieldDelimiter(v string) {
+	c.FieldDelimiter = &v
+}
+
+// SetQuoteCharacter sets quote character in CSV input options
+func (c *CSVInputOptions) SetQuoteCharacter(v string) {
+	c.QuoteCharacter = &v
+}
+
+// SetQuoteEscapeCharacter sets quote escape character in CSV input options
+func (c *CSVInputOptions) SetQuoteEscapeCharacter(v string) {
+	c.QuoteEscapeCharacter = &v
+}
+
+// SetComments sets comments character in CSV input options
+func (c *CSVInputOptions) SetComments(v string) {
+	c.Comments = &v
 }
 
 // CSVOutputOptions csv output specific options
 type CSVOutputOptions struct {
-	QuoteFields          CSVQuoteFields `xml:",omitempty"`
-	RecordDelimiter      string
-	FieldDelimiter       string `xml:",omitempty"`
-	QuoteCharacter       string `xml:",omitempty"`
-	QuoteEscapeCharacter string `xml:",omitempty"`
+	QuoteFields          *CSVQuoteFields `xml:"QuoteFields"`
+	RecordDelimiter      *string         `xml:"RecordDelimiter"`
+	FieldDelimiter       *string         `xml:"FieldDelimiter"`
+	QuoteCharacter       *string         `xml:"QuoteCharacter"`
+	QuoteEscapeCharacter *string         `xml:"QuoteEscapeCharacter"`
+}
+
+// SetQuoteFields sets quote field parameter in CSV output options
+func (c *CSVOutputOptions) SetQuoteFields(v CSVQuoteFields) {
+	c.QuoteFields = &v
```

Comment:
The best approach is to have something like a placeholder element that allows us to signify the empty value. Implement a MarshalXML which skips this. 

We should do this because Go, unfortunately, lacks proper support for default values in XML marshaling, but it also doesn't mean that we should resort to wrong things like `*string` style.

Reply:
> The best approach is to have something like a placeholder element that allows us to signify the empty value. Implement a MarshalXML which skips this.

Yes will do. We can also avoid breaking the API that way

Reply:
Well, there are already some pointers in the old code, so breaking the API is inevitable.

I am doing the change only for JSON/CSV input & output options to avoid breaking the API for now.

Reply:
Which API are we breaking? @vadmeste

Reply:
> Which API are we breaking? @vadmeste

The current PR does not break any API.

But to have better consistent API which needs to send another PR to:
- remove all pointers (there are some pointers style in the old code)
- make the code better (because the current one is not optimized because it tries to not break the API)